src/main/kotlin/com/cognifide/gradle/aem/common/instance/Ims.kt

```diff
@@ -0,0 +1,97 @@
+package com.cognifide.gradle.aem.common.instance
+
+import com.cognifide.gradle.aem.AemExtension
+import com.cognifide.gradle.aem.common.instance.service.ims.AccessObject
+import com.cognifide.gradle.aem.common.instance.service.ims.Secret
+import io.jsonwebtoken.Jwts
+import io.jsonwebtoken.SignatureAlgorithm
+import org.bouncycastle.openssl.PEMKeyPair
+import org.bouncycastle.openssl.PEMParser
+import org.bouncycastle.openssl.jcajce.JcaPEMKeyConverter
+import java.io.*
+import java.security.KeyFactory
+import java.security.spec.PKCS8EncodedKeySpec
+
+/**
+ * Generates Bearer token based on the credentials fetched from AEMaaCS
+ * by communicating with Adobe Identity Management Services.
+ */
+class Ims(private val aem: AemExtension) {
```

Comment:
this class looks much better after our refactorings; do you remember the procedural style you got initially ?:D 

now it is easy readable by human

Reply:
yeah much better :D